runner/src/main.rs

```diff
@@ -1379,14 +1379,41 @@ fn run_cargo_fmt(mode: FormatMode) -> Step {
     }
 }
 
-fn run_cargo_test() -> Step {
+fn run_cargo_test(cleanup: bool) -> Step {
     Step::Multiple {
         name: "cargo test".to_string(),
         steps: crate_manifest_files()
             .map(to_string)
-            .map(|entry| Step::Single {
-                name: entry.clone(),
-                command: Cmd::new("cargo", &["test", &format!("--manifest-path={}", &entry)]),
+            .map(|entry| {
+                if cleanup {
+                    Step::Multiple {
+                        name: entry.clone(),
+                        steps: vec![
+                            Step::Single {
+                                name: "run".to_string(),
+                                command: Cmd::new(
+                                    "cargo",
+                                    &["test", &format!("--manifest-path={}", &entry)],
+                                ),
+                            },
+                            Step::Single {
+                                name: "cleanup".to_string(),
+                                command: Cmd::new(
+                                    "rm",
+                                    &["-rf", &entry.replace("Cargo.toml", "target")],
+                                ),
+                            },
+                        ],
+                    }
+                } else {
+                    Step::Single {
+                        name: entry.clone(),
+                        command: Cmd::new(
+                            "cargo",
+                            &["test", &format!("--manifest-path={}", &entry)],
+                        ),
+                    }
```

Comment:
I think this (or at least the Cmd part) is the same in both, right? So maybe extract to a separate variable.

Reply:
Made a closure that takes the name of the step as an argument.